### PR TITLE
Convert mod_articles_popular to new structure

### PR DESCRIPTION
**Pull request by carlitorweb**
### Summary of Changes
This PR converts mod_articles_popular to the new structure

### Testing Instructions
- Use Joomla 4.3
- Create an instance of the module Articles - Most Read. Note how it works.
- Apply patch
- Delete administrator/cache/autoload_psr4.php
- Refresh the page, make sure the module still displays the same as before.

### CODE REVIEW

**Comment by carlitorweb** [comment]
@chmst I using the wrong branch?

**Comment by chmst** [comment]
@carlitorweb my apologies, was not intended to remove the label.

**Comment by chmst** [comment]
I have tested this item :white_check_mark: successfully on 130e77bb6c9c109670b12327bfafd4d55699e3cd<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39539">issues.joomla.org/tracker/joomla-cms/39539</a>.</sub>

**Comment by viocassel** [comment]
I have tested this item :white_check_mark: successfully on 130e77bb6c9c109670b12327bfafd4d55699e3cd<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39539">issues.joomla.org/tracker/joomla-cms/39539</a>.</sub>

File: modules/mod_articles_popular/src/Helper/ArticlesPopularHelper.php [hunk 1]
```diff
@@ -24,78 +31,152 @@
...
+     * Prepare the article before render.
+     *
+     * @param   object     $item   The article to prepare
+     * @param   \stdClass  $params  The model item
+     *
+     * @return  object
+     *
+     * @since   __DEPLOY_VERSION__
+     */
+    protected static function prepareItem($item, $params): object
```
Comment (laoneo) [inline modules/mod_articles_popular/src/Helper/ArticlesPopularHelper.php]:
```suggestion
    private function prepareItem($item, $params): object
```

File: modules/mod_articles_popular/src/Helper/ArticlesPopularHelper.php [hunk 2]
```diff
@@ -24,78 +31,152 @@
...
+            $itemParams->authorised = Access::getAuthorisedViewLevels($app->getIdentity()->get('id'));
+            $itemParams->access     = $access;
 
-            if ($access || \in_array($item->access, $authorised)) {
-                // We know that user has the privilege to view the article
-                $item->link = Route::_(RouteHelper::getArticleRoute($item->slug, $item->catid, $item->language));
-            } else {
-                $item->link = Route::_('index.php?option=com_users&view=login');
+            foreach ($articlesModel->getItems() as $item) {
+                $items[] = static::prepareItem($item, $itemParams);
```
Comment (laoneo) [inline modules/mod_articles_popular/src/Helper/ArticlesPopularHelper.php]:
```suggestion
                $items[] = $this->prepareItem($item, $itemParams);
```

**Review by laoneo — CHANGES_REQUESTED** [review]

**Comment by laoneo** [comment]
I would like to avoid static functions, so I made some suggestions. Can you please review them @carlitorweb ?

File: modules/mod_articles_popular/src/Helper/ArticlesPopularHelper.php [hunk 3]
```diff
@@ -24,78 +31,152 @@
...
+     * @param   SiteApplication  $app     The current application.
+     *
+     * @return  object[]
+     *
+     * @since   __DEPLOY_VERSION__
+     */
+    public function getArticles(Registry $moduleParams, SiteApplication $app)
     {
-        $app = Factory::getApplication();
+        $cacheKey = md5(serialize(array ($moduleParams->toString(), $this->module->module, $this->module->id)));
```
Comment (Quy) [inline modules/mod_articles_popular/src/Helper/ArticlesPopularHelper.php]:
```suggestion
        $cacheKey = md5(serialize([$moduleParams->toString(), $this->module->module, $this->module->id]));
```

**Review by laoneo — APPROVED** [review]

**Comment by laoneo** [comment]
Can we get here just one more test to ensure the recent cosmetic changes are correct?

**Comment by HLeithner** [comment]
thanks

**Pull request merged**